src/TurbulenceConvection/closures/entr_detr.jl

```diff
@@ -144,12 +145,89 @@ function compute_entr_detr!(
     end
 end
 
+function compute_entr_detr!(
+    state::State,
+    grid::Grid,
+    edmf::EDMFModel,
+    εδ_closure::PiGroupsDetrModel,
+    param_set::APS,
+)
+    FT = float_type(state)
+    grav = TCP.grav(param_set)
+    R_d = TCP.R_d(param_set)
+
+    N_up = n_updrafts(edmf)
+    aux_tc = center_aux_turbconv(state)
+    aux_up = center_aux_updrafts(state)
+    aux_en = center_aux_environment(state)
+    aux_gm = center_aux_grid_mean(state)
+    aux_en_f = face_aux_environment(state)
+    prog_up_f = face_prog_updrafts(state)
+    w_up_c = aux_tc.w_up_c
+    w_en_c = aux_tc.w_en_c
+
+    Ic = CCO.InterpolateF2C()
+
+    @inbounds for i in 1:N_up
+        # TODO: should we interpolate in local or covariant basis?
+        @. w_up_c = Ic(wcomponent(CCG.WVector(prog_up_f[i].w)))
+        @. w_en_c = Ic(wcomponent(CCG.WVector(aux_en_f.w)))
+
+        # updraft top height
+        H_up = compute_plume_scale_height(grid, state, edmf.H_up_min, i)
+
+        @inbounds for k in real_center_indices(grid)
+            if aux_up[i].area[k] > 0.0
+
+                z = grid.zc.z[k]
+
+                a_up = aux_up[i].area[k] # updraft area fraction
+                #b_up = aux_up[i].buoy[k] # updraft buoyancy
+                RH_up = aux_up[i].RH[k] # updraft relative humidity
+
+                a_en = aux_en.area[k] # environment area fraction
+                #b_en = aux_en.buoy[k] # environment buoyancy
+                RH_en = aux_en.RH[k] # environment relative humidity
+                tke_en = aux_en.tke[k] # environment tke
+
+                w_up = w_up_c[k] # updraft vertical velocit at cell centers
+                w_en = w_en_c[k] # environment vertical velocity at cell centers
+
+                #w_d = get_wstar(bflux)
+                T_gm = aux_gm.T[k]
+
+                #Π_1 = z * (b_up - b_en) / ((w_up - w_en)^2 + w_d^2)
+                Π_2 =
```

Comment:
```suggestion
                Π_2 = (tke_gm - a_en * tke_en) / (tke_gm + eps(FT))
```

Reply:
Π_2 was changed to this some time ago (which is a variant of the longer Π_2 form we discussed before). All calibrations (including the one for detrainment below) are with this version of Π_2 suggested here, which matched TC.jl